scripts/charm_release.py

```diff
@@ -139,37 +148,39 @@ def process_track(bundle_charms: list[str], track: str, dry_run: bool, priority_
     stable_channel = f"{track}/stable"
     k8s_operator_bundle = charmhub.Bundle("k8s-operator")
     at_least_one_charm_in_candidate = False
-    try:
-        for charm in bundle_charms:
-            print(f"Getting revisions for {charm} charm on track {track}")
+    for charm in bundle_charms:
+        print(f"Getting revisions for {charm} charm on track {track}")
+        try:
             candidate_revision_matrix = charmhub.get_revision_matrix(
                 charm, candidate_channel
             )
-            print(f"Channel {candidate_channel} revisions:")
-            print(candidate_revision_matrix)
+        except HTTPError:
+            log.exception(f"failed to get candidate revision matrix for charm {charm} channel {candidate_channel}")
+            return ProcessState.PROCESS_CI_FAILED
+        print(f"Channel {candidate_channel} revisions:")
+        print(candidate_revision_matrix)
```

Comment:
i mean, you don't HAVE to change these but

```suggestion
        log.info("Channel %s revisions:\n %s", candidate_channel, candidate_revision_matrix)
```